pkg/backend/httpstate/stack.go

```diff
@@ -128,9 +135,64 @@ func newStack(apistack apitype.Stack, b *cloudBackend) Stack {
 		tags:             apistack.Tags,
 		b:                b,
 		// We explicitly allocate the snapshot on first use, since it is expensive to compute.
+		configSource: configSource,
 	}
 }
-func (s *cloudStack) Ref() backend.StackReference                { return s.ref }
+func (s *cloudStack) Ref() backend.StackReference { return s.ref }
+
+func (s *cloudStack) ConfigSource() backend.StackConfigSource {
+	return s.configSource
+}
+
+func (s *cloudStack) Load(ctx context.Context, project *workspace.Project) (*workspace.ProjectStack, error) {
+	if s.configSource != backend.StackConfigSourceRemote {
+		return nil, errors.New("stack config source is not cloud")
+	}
+	stackID, err := s.b.getCloudStackIdentifier(s.ref)
+	if err != nil {
+		return nil, err
+	}
+	stack, err := s.b.client.GetStack(ctx, stackID)
+	if err != nil {
+		return nil, err
+	}
+	if stack.Config == nil {
+		return nil, errors.New("stack config is no longer available from the cloud")
+	}
+	projectStack := &workspace.ProjectStack{
+		Environment:     workspace.NewEnvironment([]string{stack.Config.Environment}),
+		SecretsProvider: stack.Config.SecretsProvider,
+		EncryptedKey:    stack.Config.EncryptedKey,
+		EncryptionSalt:  stack.Config.EncryptionSalt,
+	}
+	return projectStack, nil
+}
+
+func (s *cloudStack) Save(ctx context.Context, projectStack *workspace.ProjectStack) error {
```

Comment:
Doesn't this or something need to handle updating the ESC environment?

Reply:
Nope, we just error for now. The cli command well direct them to use an esc command directly instead